src/xunit.assert/Asserts/Sdk/AssertEqualityComparer.cs

```diff
@@ -149,6 +158,36 @@ public bool Equals(T x, T y)
             return dictionaryYKeys.Count == 0;
         }
 
+        bool? CheckIfSetsAreEqual(T x, T y, TypeInfo typeInfo)
+        {
+            if (!IsSet(typeInfo))
+                return null;
+
+            var enumX = x as IEnumerable;
+            var enumY = y as IEnumerable;
+            if (enumX == null || enumY == null)
+                return null;
+
+            var objectEnumX = enumX.Cast<object>();
+            var objectEnumY = enumY.Cast<object>();
+            if (objectEnumX.Count() != objectEnumY.Count())
+                return false;
+
+            var setX = new HashSet<object>(objectEnumX, objectComparer);
```

Comment:
T here is not the object type, it's ISet<M>. I couldn't find a way to cast the the sets to ISet<M> and compare them using ISet's comparer, that's why I used the object comparer (or the inner comparer if there is one). The result should be the same. Plus, I wanted to allow using a custom comparer in the future.